packages/core/helper-plugin/lib/src/components/RelationInput/RelationInput.js

```diff
@@ -0,0 +1,179 @@
+              inputId={id}
+              isSearchable
+              isClear
+              onChange={onRelationAdd}
+              onInputChange={onSearch}
+              onMenuClose={onRelationOpen}
+              onMenuOpen={onRelationClose}
+              onMenuScrollToBottom={onSearchNextPage}
+              placeholder={placeholder}
+            />
+          </>
+        }
+        loadMore={
+          !disabled && (
+            <TextButton onClick={() => onRelationLoadMore()} startIcon={<Refresh />}>
+              {labelLoadMore}
+            </TextButton>
+          )
+        }
+      >
+        <RelationList height={listHeight}>
+          {relations.isSuccess &&
+            relations.data.pages.flatMap((relation) => {
+              const { isDraft, href, title, id } = relation;
+              const badgeColor = isDraft ? 'secondary' : 'success';
+
+              return (
+                <RelationItem
+                  disabled={disabled}
+                  key={`relation-${name}-${id}`}
+                  endAction={
+                    <button
+                      disabled={disabled}
+                      type="button"
+                      onClick={() => onRelationRemove(relation)}
+                    >
+                      <Icon width="12px" as={Cross} />
+                    </button>
+                  }
+                >
+                  <Box paddingTop={1} paddingBottom={1}>
+                    {href ? (
+                      <BaseLink disabled={disabled} href={href}>
+                        {title}
+                      </BaseLink>
+                    ) : (
+                      title
+                    )}
+                  </Box>
+
+                  {isDraft !== undefined && (
+                    <Badge
+                      borderSize={1}
+                      borderColor={`${badgeColor}200`}
+                      backgroundColor={`${badgeColor}100`}
+                      textColor={`${badgeColor}700`}
+                    >
+                      {isDraft ? 'Draft' : 'Published'}
+                    </Badge>
+                  )}
+                </RelationItem>
+              );
+            })}
+        </RelationList>
+        <Box paddingTop={2}>
+          <FieldHint />
+          <FieldError />
+        </Box>
+      </Relation>
+    </Field>
+  );
+};
+
+const ReactQueryRelationResult = PropTypes.shape({
+  data: PropTypes.shape({
+    pages: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        isDraft: PropTypes.bool,
+        href: PropTypes.string,
+        title: PropTypes.string.isRequired,
+      })
+    ),
+  }),
+  isLoading: PropTypes.bool.isRequired,
+  isSuccess: PropTypes.bool.isRequired,
+});
+
+const ReactQuerySearchResult = PropTypes.arrayOf(
+  PropTypes.shape({
+    label: PropTypes.string,
+    value: {
+      id: PropTypes.number,
+      name: PropTypes.string,
+      publishedAt: PropTypes.string,
+    },
+  })
+);
```

Comment:
With https://github.com/strapi/strapi/pull/14092/commits/302765d7ddfa2732e8f364dd10c6ebc754005c8d you can expect a very similar shape here now too:

```js
const ReactQuerySearchResult = PropTypes.shape({
  data: PropTypes.shape({
    pages: PropTypes.arrayOf(
      PropTypes.shape({
        isDraft: PropTypes.bool,
        mainField: PropTypes.oneOfType([PropTypes.string, PropTypes.number,])
        // plus all fields the API returns, but we don't need to type
      })
    ),
  }),
  isLoading: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
});
```